pkg/app/piped/analysisprovider/metrics/provider.go

```diff
@@ -28,11 +28,22 @@ var (
 // Provider represents a client for metrics provider which provides metrics for analysis.
 type Provider interface {
 	Type() string
+
 	// Evaluate runs the given query against the metrics provider,
 	// and then checks if the results are expected or not.
 	// Returns the result reason if non-error occurred.
 	// The first value "expected" must be false if err isn't nil.
+	// TODO: Do not evaluate data points by Analysis providers
+	//   Instead, the executor should do that by using SelectPoints().
 	Evaluate(ctx context.Context, query string, queryRange QueryRange, evaluator Evaluator) (expected bool, reason string, err error)
+
+	// SelectPoints gives back data points within the given range.
+	SelectPoints(ctx context.Context, query string, queryRange QueryRange) (points []DataPoint, err error)
```

Comment:
nit: `QueryPoints` and `queryRange` -> `range`

Reply:
Thanks. Does it mean `type QueryRange struct` should be `Range`?

Reply:
Ah, you mean it should be `QueryPoints ` instead of `SelectPoints`.

Reply:
I was talking about the name, not the type. w

Reply:
`SelectPoints` -> `QueryPoints`
`queryRange` -> `range`

Reply:
Honestly, the word `range` isn't available in Go because of the reserved word.

Reply:
Alright 👍

Reply:
Ah right. So that is fine. 👍